### PR TITLE
Test queues

### PR DESCRIPTION
**Pull request by novichkovv**
This version works quite stable, I want to test on prod

### CODE REVIEW

File: protected/classes/push_class.php [hunk 1]
```diff
@@ -13,6 +13,9 @@ public static function sendPushes() {
         while (($pushes = $q->Get(self::NUM_READ_MESSAGES)) != array()) {
             foreach ($pushes as $push) {
                 $push['message']['id'] = time() . rand() .rand();
+                if($push['message']['last']) {
```
Comment (mosheka) [inline protected/classes/push_class.php]: Please notice that may not keep order, and actually the last message may be processed before other.
Why do we need it? We close the task_queue in task manager

File: protected/classes/task_class.php [hunk 2]
```diff
@@ -78,7 +78,11 @@ function enrichUsers($users) {
                 'queue_id' => self::$task['queue_id'],
                 'create_date' => gmdate('Y-m-d H:i:s')
             ];
-            self::writeLog('TASK',$user['id'] .CS . $user['ip'] . CS . $task['id'] . CS . $unit['ua'] . CS . $unit['campaign_id']);
+            if($k === count($users) - 1) {
+                $unit['last'] = 1;
```
Comment (mosheka) [inline protected/classes/task_class.php]: it better to have $count_users = count($users)
And compare to static value

File: protected/classes/task_class.php [hunk 3]
```diff
@@ -54,7 +54,7 @@ public static function createUnits($task) {
     function enrichUsers($users) {
         $units = [];
         $pushes = [];
```
Comment (mosheka) [inline protected/classes/task_class.php]: $count_users = count($users)

File: protected/classes/task_class.php (hunk 2 above)
Comment (mosheka) [reply protected/classes/task_class.php]: (if $k == ($count_users - 1)) {

File: protected/classes/push_class.php (hunk 1 above)
Comment (novichkovv) [reply protected/classes/push_class.php]: It's not important for this to be absolutely accurate, it's for UI to be informative, that's what Kobi asked. I'll show you if you need

File: protected/classes/task_class.php (hunk 2 above)
Comment (novichkovv) [reply protected/classes/task_class.php]: ok

**Pull request merged**